File: problems/001-anagram-detection/kotlin/Optimized.kt
class Optimized : Solution {
    override fun solve(c: Map<String, Any?>): Any? {
        val s = c["s"] as String
        val t = c["t"] as String
        if (s.length != t.length) return false
        val count = IntArray(256)
        for (ch in s) count[ch.code]++
        for (ch in t) if (--count[ch.code] < 0) return false
        return true
    }
}

File: problems/001-anagram-detection/swift/optimized.swift
import Foundation

func optimized(_ c: [String: Any]) -> Any {
    let s = c["s"] as! String, t = c["t"] as! String
    if s.count != t.count { return false }
    var count = [Character: Int]()
    for ch in s { count[ch, default: 0] += 1 }
    for ch in t {
        count[ch, default: 0] -= 1
        if count[ch]! < 0 { return false }
    }
    return true
}
